src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchResponseMapper.java

```diff
@@ -4,96 +4,93 @@
 
-        var properties =  new ArrayList<Property>();
-        result.getEntity().get(universe.getName())
-                .forEach((key, value) -> properties.add(new Property(key, (String) value)));
+        if ("SUCCESS".equals(result.getStatus())) {
+            var matches = result.getMatch().stream()
+                    .map(match -> createMatchesToProperty(universe, match, universe.getIdField(), true))
+                    .collect(Collectors.toList());
 
-        properties.add(new Property(GoldenRecordConstants.SOURCE_ID_FIELD, result.getIdResource()));
-        properties.add(new Property(FuzzyMatchDetialsConstants.FUZZY_MATCH_DETAILS, (MObject) null));
-        properties.addAll(object.getProperties());
+            propertiesMatched = new Property(FuzzyMatchDetailsConstants.MATCH, matches);
 
-        object.setProperties(properties);
+            var duplicates = result.getDuplicate().stream()
+                    .map(duplicate -> createMatchesToProperty(universe, duplicate, universe.getIdField(), true))
+                    .collect(Collectors.toList());
+
+            propertiesDuplicated = new Property(FuzzyMatchDetailsConstants.DUPLICATE, duplicates);
 
-        if ("SUCCESS".equals(result.getStatus())) {
-            result.getMatch().forEach(match -> addMatchesToProperty(propertiesMatched, universe, match, universe.getIdField(), true));
-            result.getDuplicate().forEach(match -> addMatchesToProperty(propertiesDuplicated, universe, match, universe.getIdField(), true));
         } else if ("ALREADY_LINKED".equals(result.getStatus())) {
-            var entityH = result.getEntity().get(universe.getName());
-            addAlreadyLinked(propertiesAlreadyLinked, universe, entityH, universe.getIdField());
+            var entityLinked = result.getEntity().get(universe.getName());
+            propertiesAlreadyLinked = new Property(FuzzyMatchDetailsConstants.ALREADY_LINKED,
+                    Arrays.asList(createAlreadyLinked(universe, entityLinked, universe.getIdField())));
         }
 
-        return object;
-    }
+        var properties = result.getEntity().get(universe.getName()).entrySet().stream()
+                .filter(entry -> entry.getValue() instanceof String)
+                .map(entry -> new Property(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+
+        properties.add(new Property(GoldenRecordConstants.SOURCE_ID_FIELD, result.getIdResource()));
+        properties.add(new Property(FuzzyMatchDetailsConstants.FUZZY_MATCH_DETAILS, (MObject) null));
 
 
-    private static void addAlreadyLinked(Property alreadyLinkedProperty, Universe universe, Map<String, Object> entity, String idField){
-        var object = new MObject(universe.getId().toString() + " match");
-        object.setExternalId(entity.get(idField).toString());
+        properties.addAll(Arrays.asList(propertiesMatched, propertiesDuplicated, propertiesAlreadyLinked));
```

Comment:
If you're adding 3 different collections, it's probably best to just call `addAll` three times rather than wrapping them in another list. It'll be more explicit and the performance regression probably doesn't matter in this case